lib/options-manager.js

```diff
@@ -44,6 +55,43 @@ const DEFAULT_CONFIG = {
 	}
 };
 
+/**
+ * Define the rules that are enabled only for specific version of Node, based on `engines.node` in package.json or the `node-engine` option.
+ *
+ * The keys ae rule names and the values are an Object with a valid semver (`4.0.0` is valid `4` is not) as keys and the rule configuration as values.
+ * Each entry define the rule configuration and the minimum Node version for which to set it.
+ * The entry with the highest version that is compliant with the `engines.node`/`node-engine` range will be used.
+ *
+ * @type {Object}
+ *
+ * @example
+ * ```javascript
+ * {
+ * 	'plugin/rule': {
+ * 		'6.0.0': ['error', {prop: 'node-6-conf'}],
+ *		'8.0.0': ['error', {prop: 'node-8-conf'}]
+ * 		}
+ * }
+ *```
+ * With `engines.node` set to `>=4` the rule `plugin/rule` will not be used.
+ * With `engines.node` set to `>=6` the rule `plugin/rule` will be used with the config `{prop: 'node-6-conf'}`.
+ * With `engines.node` set to `>=8` the rule `plugin/rule` will be used with the config `{prop: 'node-8-conf'}`.
+ */
+const ENGINE_RULES = {
+	'promise/prefer-await-to-then': {
+		'8.0.0': 'error'
+	},
+	'prefer-rest-params': {
+		'6.0.0': 'error'
+	},
+	'prefer-spread': {
+		'5.0.0': 'error'
+	},
+	'prefer-destructuring': {
+		'6.0.0': ['error', {array: true, object: true}, {enforceForRenamedProperties: true}]
+	}
+};
```

Comment:
Can you include https://github.com/sindresorhus/eslint-config-xo/blob/3f8179657736c4c078759416509ca9698e5321e6/esnext.js#L16 too?

Reply:
Id didn't include it becasue it's deprecated. See https://eslint.org/docs/rules/prefer-reflect

Reply:
Ah, I didn't realize. We can eventually fix that: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/142

Reply:
I think there is a bunch of rules to add from `eslint-plugin-unicorn` and some of them should apply only for specific Node.js version. But that might be better to do that in a different PR.